src/compiler/checker.ts

```diff
@@ -9348,7 +9348,7 @@ namespace ts {
                             return anyType;
                         }
                         checkNoTypeArguments(node);
-                        return anyType;
+                        return !noImplicitAny ? anyType : undefined;

```

Comment:
these were the minimal change, but they could be negated or split up if that seems clearer